### PR TITLE
Piechart: Implements series override -> hide in area for the legend or tooltip

### LINKED ISSUE #49972: Pie chart: Can't hide items from legend or tooltip
**What happened**:  
Made a pie chart with multiple series, wanted to hide some. Series override -> Hide in Area only works for the viz, not the legend or tooltip.

image

**What you expected to happen**:  
Clicking the buttons to hide in tooltip or legend hides the items in those areas.

**How to reproduce it (as minimally and precisely as possible)**:  
Make a pie chart with any data, try to hide it through series override.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v9.0.0-beta2 (3ed722bb5c)
- Data source type & version: Dashboard, TimescaleDB, any
- OS Grafana is installed on: Raspbian, Podman container
- User OS & Browser: Fedora, Chrome
- Grafana plugins: Stock from Docker Hub
- Others: Happened on Grafana 8.x as well.

### PR DESCRIPTION
**Pull request by daniellee**
**What this PR does / why we need it**:

Missing functionality in the pie chart panel for the Series override -> Hide in area. 

Shows hiding a country first from the legend and then from the multi tooltip.

Piechart-hide

**Which issue(s) this PR fixes**:

Fixes #49972

**Special notes for your reviewer**:

Not sure if there is a way to avoid duplication. There are two places where this series override is implemented already in the PlotLegend and the TooltipPlugin.

I looked at the [PlotLegend code](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/uPlot/PlotLegend.tsx#L53) for the legend and tried to keep it similar. The tooltip logic is simpler than the [TooltipPlugin implementation](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx#L215).

### CODE REVIEW

File: public/app/plugins/panel/piechart/PieChartPanel.tsx [hunk 1]
```diff
@@ -93,14 +93,20 @@ function getLegend(props: Props, displayValues: FieldDisplay[]) {
...
-    .map<VizLegendItem>((value, idx) => {
-      const hidden = value.field.custom.hideFrom.viz;
+    .map<VizLegendItem | undefined>((value, idx) => {
+      const hideFromViz = value.field.custom.hideFrom.viz;
+      const hideFromLegend = value.field.custom?.hideFrom?.legend;
+
+      if (hideFromLegend) {
+        return undefined;
+      }
+
```
Comment (ryantxu) [inline public/app/plugins/panel/piechart/PieChartPanel.tsx:103]: Maybe
```suggestion
      const hideFrom = (value.field.custom?.hideFrom ?? {}) as HideSeriesConfig;
      if (hideFrom.legend) {
        return undefined;
      }

      const hideFromViz = Boolean(hideFrom.viz)
```
it has the hideFrom time involved, so a little nicer

Comment (daniellee) [reply public/app/plugins/panel/piechart/PieChartPanel.tsx]: That is nicer. Couldn't commit your suggestion as it was missing the import for HideSeriesConfig - so added it manually.

Comment (daniellee) [reply public/app/plugins/panel/piechart/PieChartPanel.tsx]: @ryantxu looks like https://github.com/grafana/grafana/pull/50229 broke the main branch. Tons of errors in betterer that are nothing to do with this PR.

**Comment by daniellee** [comment]
@ryantxu looks like https://github.com/grafana/grafana/pull/50229 broke the main branch. Tons of errors in betterer that are nothing to do with this PR.

File: public/app/plugins/panel/piechart/PieChartPanel.tsx (hunk 1 above)
Comment (daniellee) [reply public/app/plugins/panel/piechart/PieChartPanel.tsx]: Actually ran into a betterer check for your suggestion - it is adding a type assertion:

image

**Review by ryantxu — APPROVED** [review]
Looks good!  now all the things work :) 


I still don't get how/why using `... as TypeWeKnow` is worse than using the untyped existing `any`, but that is a much more complicated discussion

**Comment by adamxyzxyz** [comment]
hey guys what version is this live in?

**Pull request merged**